### DISCUSSION TITLE
Theming questions

### DISCUSSION THREAD

**Discussion by IceReaper** [BlazorFluentUI/BlazorFluentUI#357] [General]
Is there any particular reason some css uses --palette-* variables for background colors instead of the semantic colors?
Basicaly when implementing a dark ui, it requires to define Black as a white color and White as a black color, which is extremely ugly as it might cause huge confusions.

If there is no particular reason i would suggest the following:
- Replacing all direct references of --palette-* with semantic variables.
- Get rid of all the Theme* Neutral* properties from IPalette, and make the palette implementation define which theme colors exist, and add the required IPalette type in the ISemanticColors implementations, except where stuff is required to be a specific color. Like red for an error.
- Leave in all --palette-<colorname>. red will be red, black will be black, blue will be blue, etc.

This way one could setup an IPalette with as many colors required for the theme, and properly assign them using the semantics.

**Comment by vnbaaij** [comment]
I don't know why those choices were made back then, but what you say makes sense and I think it concurs with https://github.com/microsoft/fluentui/wiki/Theming#what-are-theme-slots. 

You seem to have a good idea on how this can be changed /solved. Why not submit a PR? Happy to review and merge!

**Comment by limefrogyank** [comment]
We copied their theme when they themselves hadn't swapped to using semantic for everything.

Lee McPherson


________________________________
